osu.Game.Rulesets.Osu/Mods/OsuModApproachCircle.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    internal class OsuModApproachCircle : Mod, IApplicableToDrawableHitObjects
+    {
+        public override string Name => "Approach Circle";
+        public override string Acronym => "AC";
+        public override string Description => "Never trust the approach circles...";
+        public override double ScoreMultiplier => 1;
+        public override IconUsage? Icon { get; } = FontAwesome.Regular.Circle;
+
+        [SettingSource("Easing", "Change the easing type of the approach circles.", 0)]
+        public Bindable<ReadableEasing> BindableEasing { get; } = new Bindable<ReadableEasing>();
+
+        [SettingSource("Scale the size", "Change the factor of the approach circle scale.", 1)]
+        public BindableFloat Scale { get; } = new BindableFloat
+        {
+            Precision = 0.1f,
+            MinValue = 2,
+            MaxValue = 10,
+            Default = 4,
+            Value = 4
+        };
+
+        public void ApplyToDrawableHitObjects(IEnumerable<DrawableHitObject> drawables)
+        {
+            drawables.ForEach(drawable =>
+            {
+                drawable.ApplyCustomUpdateState += (drawableHitObj, state) =>
+                {
+                    if (!(drawableHitObj is DrawableHitCircle hitCircle)) return;
+
+                    var obj = hitCircle.HitObject;
+
+                    hitCircle.BeginAbsoluteSequence(obj.StartTime - obj.TimePreempt, true);
+                    hitCircle.ApproachCircle.ScaleTo(Scale.Value);
+
+                    hitCircle.ApproachCircle.FadeIn(Math.Min(obj.TimeFadeIn, obj.TimePreempt));
+
+                    hitCircle.ApproachCircle.ScaleTo(1f, obj.TimePreempt, (Easing)BindableEasing.Value);
+
+                    hitCircle.ApproachCircle.Expire(true);
+                };
+            });
+        }
+
+        internal enum ReadableEasing
+        {
+            Accelerate = 2,
+            Accelerate2 = 6,
+            Accelerate3 = 12,
+            AccelerateInAfterDeceleraingeOut = 29,
+            CasualBounces = 32,
+            CasualBouncesWhileAccelerating = 24,
+            Decelerate = 1,
+            DecelerateAfterAccelerating = 8,
+            Default = 0,
+        }
```

Comment:
* It's bad practice to rely directly on the underlying numerical values of the `Easing` enum. If someone adds another enum value in between the existing ones in the framework it'll silently break this mapping since the numerical values depend on member order in code. Should try to reference the values directly through casting if anything.
* The names are only *slightly* better and IMO demonstrate just how arbitrary allowing this setting for easing curve is. How does _Accelerate_ differ from _Accelerate 2_ and _Accelerate 3_? What are _casual bounces_? With other mods, you pretty much know what you're going to get with them. But this one - it's tweaking implementation details of an animation.

Reply:
This would make sense if we had some sort of a visualizer (just a hitcircle visualizing the currently chosen animation) next to the setting dropdown - but I guess it's not really possible in the current mod menu, nor is it feasible to really try to implement given its scope.

Reply:
Remove all numbers here and reorder so "Default" is at the top. Make a mapping function. Also what is a "casual" bounce?